File: lclang/src/test/java/lclang/libs/TestLibrary.java
package lclang.libs;

import lclang.types.Types;

public class TestLibrary extends Library {
    public final StringBuilder output;

    public TestLibrary(StringBuilder output) {
        super("test");
        this.output = output;
        globals.put("print", voidMethod((caller, args) ->
                output.append(args.get(0).toString(caller)), Types.ANY).asValue());
        globals.put("println", voidMethod((caller, args) ->
                output.append(args.get(0).toString(caller)).append("\r\n"), Types.ANY).asValue());
    }
}

File: lclang/src/test/kotlin/lclang/libs/TestLibrary.kt
package lclang.libs

import lclang.types.Types
import lclang.void

class TestLibrary(output: StringBuilder): Library("test") {
    init {
        globals["println"] = void(Types.ANY) {
            output.append("${it[0]}\r\n")
        }

        globals["print"] = void(Types.ANY) {
            output.append(it[0])
        }
    }
}